File: ios/Sources/VaultSync/Domain/UseCases/StartSyncUseCase.swift
import Foundation

public final class StartSyncUseCase: Sendable {
    private let syncController: any SyncController

    public init(syncController: any SyncController) {
        self.syncController = syncController
    }

    public func execute() async throws -> SyncReport {
        try await syncController.startSync()
    }
}

File: android/app/src/main/java/com/vaultsync/domain/usecase/StartSyncUseCase.kt
package com.vaultsync.domain.usecase

import com.vaultsync.domain.model.SyncReport
import com.vaultsync.domain.repository.SyncController

class StartSyncUseCase(
    private val syncController: SyncController
) {
    suspend operator fun invoke(): SyncReport {
        return syncController.startSync()
    }
}
